### PR TITLE
Error if there is no main() when using PROXY_TO_PTHREAD

### PR DESCRIPTION
**Pull request by kripken**

### CODE REVIEW

**Review by sbc100 — APPROVED** [review]

**Comment by kripken** [comment]
Disabled the error in relocatable builds. main() might arrive from another module there.

**Review by tlively — APPROVED** [review]

**Comment by kripken** [comment]
`other.test_metadce_minimal_pthreads` used PROXY_TO_PTHREAD without a main(). I removed the proxying part, which alters the expectations a little.

File: tests/test_other.py [hunk 1]
```diff
@@ -7094,7 +7094,7 @@ def test_metadce_minimal(self, *args):
 
   @node_pthreads
   def test_metadce_minimal_pthreads(self):
-    self.run_metadce_test('minimal.c', ['-Oz', '-sUSE_PTHREADS', '-sPROXY_TO_PTHREAD'], [], [])
+    self.run_metadce_test('minimal.c', ['-Oz', '-sUSE_PTHREADS'], [], [])
```
Comment (sbc100) [inline tests/test_other.py:7097]: But not there is no actual usage of threading in this example.. so I don't think its testing what we want it to test anymore.

Perhaps instead create a `minimal_main.c` which does have main.  Or alternatively create a `minimal_threads.c` which creates at least one thread.

Comment (kripken) [reply tests/test_other.py]: Makes sense, done.

**Review by sbc100 — APPROVED** [review]
nice!

**Pull request merged**